vxfw/vxlayout/flex2.go

```diff
@@ -0,0 +1,422 @@
+package vxlayout
```

Comment:
Note: I (mostly) left flex.go alone and placed all the new stuff in here to simplify reading the new implementation. There's only a couple of parts of flex.go that are used in here (some of the methods on FlexDirection).

If and when this is accepted, I'll remove flex.go as is and rename this file in place.